internal/api/v1beta1/app_types.go

```diff
@@ -565,9 +574,29 @@ func (app *App) DoCanary(now metav1.Time) error {
 			app.Spec.Deployments = []AppDeploymentSpec{app.Spec.Deployments[1]}
 		}
 	}
+
+	// update failure count on successful canary rollout
+	if app.Spec.Canary.FailureCount > 0 {
```

Comment:
why we should do it?

Reply:
I was thinking about consecutive failures, so idea is to reset it on successful rollouts